01_fib/alextmz/main_test.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+)
+
+func Test_main(t *testing.T) {
+	want := "1\n1\n2\n3\n5\n8\n13\n"
+
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	result := buf.String()
+
+	if result != want {
+		t.Errorf("expected %v, got %v", want, result)
+	}
+}
+
+func Test_printfib(t *testing.T) {
+
+	tests := []struct {
+		name string
+		args int
+		want string
+	}{
+		{"Print Fibonacci sequence for 0", 0, ""},
+		{"Print Fibonacci sequence for 1", 1, "1\n"},
+		{"Print Fibonacci sequence for 2", 2, "1\n1\n"},
+		{"Print Fibonacci sequence for 3", 3, "1\n1\n2\n"},
+		{"Print Fibonacci sequence for 7", 7, "1\n1\n2\n3\n5\n8\n13\n"},
+	}
+
+	for _, tr := range tests {
+		tr2 := tr
```

Comment:
just a note: this could be `tr := tr` which you'll often see in loops where you need to take a local variable. There's no need for a new name, the inner scope name can shadow the outer scope name

Reply:
Yes, I saw it could be done like that but the code can look confusing - I'm still a bit torn between trying to be idiomatic or simple/clear...